src/test/java/com/google/neighborgood/IntegrationTest.java

```diff
@@ -0,0 +1,1065 @@
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(
+                        By.xpath("//tbody[@id='need-help-body']/tr[1]/td[2]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("helper"), taskHelper);
+
+    verifyTaskDetails("//tbody[@id='need-help-body']/tr[1]/td[1]");
+  }
+
+  @Test
+  /** Test functionality of a helper abandoning a task */
+  public void _10_UserPage_AsHelper_AbandonTask() {
+    logOut("logout-href");
+    ifLaggingThenRefresh();
+    loginUser(USER_EMAIL_HELPER);
+    ifLaggingThenRefresh();
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+    goToOfferHelp();
+    // clicks on abandon task button element
+    wait.until(
+            new Function<WebDriver, WebElement>() {
+              public WebElement apply(WebDriver driver) {
+                return driver.findElement(
+                    By.xpath("//tbody[@id='offer-help-body']/tr/td[6]/button"));
+              }
+            })
+        .click();
+    // Clicking abandon triggers and alert confirmation window
+    // Driver will try to accept the alert to verify the task every second for a minute
+    for (int i = 0; i < 60; i++) {
+      try {
+        driver.switchTo().alert().accept();
+        break;
+      } catch (NoAlertPresentException e) {
+        driver.manage().timeouts().implicitlyWait(1, TimeUnit.SECONDS);
+      }
+    }
+    // Updates recentTask
+    recentTask.put("status", "OPEN");
+    backToHome();
+    ifLaggingThenRefresh();
+    // verifies that abandoned task is now the most recent task in homepage
+    verifyNewTaskHomepage();
+  }
+
+  /** Clears entities from Datastore so `mvn clean` isn't necessary before test class */
+  private static void clearAllDatastoreEntities(WebDriver driver) {
+    driver.get("http://localhost:8080/_ah/admin");
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+    WebElement entityKindSelect =
+        wait.until(
+            new Function<WebDriver, WebElement>() {
+              public WebElement apply(WebDriver driver) {
+                return driver.findElement(By.id("kind_input"));
+              }
+            });
+    // Retrieves number of Entity Kind select options to iterate through them
+    Select kindSelect = new Select(entityKindSelect);
+    List<WebElement> allEntityKinds = kindSelect.getOptions();
+    WebElement listButtonElement;
+    WebElement allKeysElement;
+    WebElement deleteButtonElement;
+    for (int j = 1; j < allEntityKinds.size(); j++) {
+      // clicks on list button
+      wait.until(
+              new Function<WebDriver, WebElement>() {
+                public WebElement apply(WebDriver driver) {
+                  return driver.findElement(By.id("list_button"));
+                }
+              })
+          .click();
+      driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
```

Comment:
We have already waited for the previous code block getting executed, why we still have to wait for 5 seconds here?

Reply:
The wait here is to give time for the response to the click to 'process' as it sends the driver to another page.